src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -0,0 +1,79 @@
+import Header from "../Header";
+import { useParams } from "react-router-dom";
+import { viewSingleArticle, getComments } from "../../utils/api";
+import { useState, useEffect } from "react";
+import "./SingleArticle.css";
+
+export default function SingleArticle() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState([]);
+  const [comments, setComments] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const articleDate = new Date(article.created_at).toString();
+
+  useEffect(() => {
+    viewSingleArticle(article_id)
+      .then((response) => {
+        setArticle(response);
+      })
+      .catch((err) => {
+        setError(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+      
+    getComments(article_id)
+      .then((response) => {
+        setComments(response);
+      })
+      .catch((err) => {
+        setError(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
```

Comment:
It looks like you've implement ticket 5 and 6 together here.

This branch should only be for ticket 5 - be careful of this in future.

As ticket 6 (CORE: View a list of comments associated with an article) is its own user story, I'd encourage you strongly to refactor any logic dealing with showing comments and split them into their own components and move the api call into a comments component.

There's now alot going on here because you've incorporated tickets 5 and 6

I wouldn't recommend sharing the `isLoading, setLoading` state and set state function for both the single article and comments, again this is a good reason to split the rendering of comments into its own component, with its own isloading pattern.